### DISCUSSION TITLE
Make Producer.Send method also accept Message[] in addition to List<Message>

### DISCUSSION THREAD

**Discussion by plewam** [rabbitmq/rabbitmq-stream-dotnet-client#338] [Ideas]
Hi All,

In my streaming system I internally work with arrays for faster access. However when I push to rmq streams I always have to convert my already existing array into a list by calling `ToList()`.

`await Producer.Send(streamMessage.ToList(), CompressionType.Gzip).ConfigureAwait(false);`

As far as I understand it, it would only make sense for the batch feature.

Not a big deal, but more a smaller performance improvement.

**Comment by Gsantomaggio** [comment]
Hi @plewam thank you for reporting the question.

We can add it, but we would do the same `ToList()` conversion unless we add another publish command:

https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/blob/cf8c927575e06c5ebad5eb0fd58fe34e67d56453/RabbitMQ.Stream.Client/Publish.cs#L33-L54


I don't know the percentage of gain in terms of performance using an array instead of a List.

**Reply by plewam** [reply]
Ah I see, as you are not altering the list you could potentially use an array instead. This article provides a nice overview why the array is faster, https://www.bytehide.com/blog/array-vs-list-csharp

The main reason simply is the way faster access speed of an array vs. a list.

**Reply by Gsantomaggio** [reply]
I agree that an array can be faster than a List. I was wondering if this change has a real impact on stream the system.

- We currently have other issues to follow, but it would be helpful if you could make the change and report some benchmark results.


- Do you have performance problems?

**Comment by Gsantomaggio** [comment]
closing! Feel free to reopen with more details
